app/src/features/surveys/observations/observations-table/configure-table/ConfigureColumnsContainer.tsx

```diff
@@ -0,0 +1,208 @@
+      (item) => item === true
+    );
+
+    let model: GridColumnVisibilityModel = {};
+
+    if (hasHiddenColumns) {
+      // Some columns currently hidden, show all columns
+      model = { ...Object.fromEntries(hideableColumns.map((column) => [column.field, false])) };
+    } else {
+      // No columns currently hidden, hide all columns
+      model = { ...Object.fromEntries(hideableColumns.map((column) => [column.field, true])) };
+    }
+
+    // Store current visibility model in session storage
+    sessionStorage.setItem(
+      getSurveySessionStorageKey(surveyId, SIMS_OBSERVATIONS_HIDDEN_COLUMNS),
+      JSON.stringify(model)
+    );
+
+    // Update the column visibility model in the context
+    observationsTableContext.setColumnVisibilityModel(model);
+  }, [hideableColumns, observationsTableContext, surveyId]);
+
+  /**
+   * Removes measurement columns from the observations table, ignoring columns that don't exist.
+   *
+   * @param {string[]} measurementColumnsToRemove The `field` names of the columns to remove
+   */
+  const onRemoveMeasurements = useCallback(
+    (measurementColumnsToRemove: string[]) => {
+      observationsTableContext.setMeasurementColumns((currentColumns) => {
+        const remainingColumns = currentColumns.filter(
+          (currentColumn) => !measurementColumnsToRemove.includes(currentColumn.colDef.field)
+        );
+
+        // Store user-added mesurement columns in local storage
+        sessionStorage.setItem(
+          getSurveySessionStorageKey(surveyId, SIMS_OBSERVATIONS_MEASUREMENT_COLUMNS),
+          JSON.stringify(remainingColumns)
+        );
+
+        return remainingColumns;
+      });
+    },
+    [observationsTableContext, surveyId]
+  );
+
+  /**
+   * Handles the addition of measurement columns to the table.
+   *
+   * @param {CBMeasurementType[]} measurements
+   * @return {*}
+   */
+  const onAddMeasurements = (measurements: CBMeasurementType[]) => {
+    if (!measurements?.length) {
+      return;
+    }
+
+    const measurementColumnsToAdd: MeasurementColumn[] = getMeasurementColumns(
+      measurements,
+      observationsTableContext.hasError
+    );
+
+    observationsTableContext.setMeasurementColumns((currentColumns) => {
+      const newColumns = measurementColumnsToAdd.filter(
+        (columnToAdd) =>
+          !currentColumns.find((currentColumn) => currentColumn.colDef.field === columnToAdd.colDef.field)
+      );
+
+      // Store user-added mesurement columns in local storage
+      sessionStorage.setItem(
+        getSurveySessionStorageKey(surveyId, SIMS_OBSERVATIONS_MEASUREMENT_COLUMNS),
+        JSON.stringify([...currentColumns, ...newColumns])
+      );
+
+      return [...currentColumns, ...newColumns];
+    });
+  };
+
+  /**
+   * On first mount, load visibility state from session storage, if it exists.
+   */
+  useEffect(() => {
+    if (JSON.stringify(observationsTableContext.columnVisibilityModel) !== '{}') {
```

Comment:
Still looking for an alternative here? Would this work?
```javascript
if (Object.keys(observationsTableContext.columnVisibilityModel).length) {
  // ...
}
```

Reply:
Yeah thats better, good call